packages/react/src/components/Card/CardTypeContent.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useMemo, useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { defaultsDeep } from 'lodash-es';
+
+import { MeterChart } from '../MeterChart';
+import { AreaChart } from '../AreaChart';
+import { StackedAreaChart } from '../StackedAreaChart';
+import { CARD_TYPES } from '../../constants/LayoutConstants';
+import { MeterChartPropTypes, SparklineChartPropTypes } from '../../constants/ChartPropTypes';
+import { settings } from '../../constants/Settings';
+import { getChartOptions } from '../../utils/cardTypeUtilityFunctions';
+
+const { iotPrefix } = settings;
+const propTypes = {
+  testId: PropTypes.string,
+  isExpanded: PropTypes.bool,
+  type: PropTypes.oneOf([
+    CARD_TYPES.METER_CHART,
+    CARD_TYPES.SPARKLINE_CHART,
+    CARD_TYPES.STACKED_AREA_CHART,
+  ]).isRequired,
+  data: PropTypes.oneOf([MeterChartPropTypes.data, SparklineChartPropTypes.data]),
+  content: PropTypes.oneOf([MeterChartPropTypes.content, SparklineChartPropTypes.content]),
+};
+
+const defaultProps = {
+  testId: 'cart-type-content',
+  isExpanded: false,
+  data: [],
+  content: null,
+};
+
+const CardTypeContent = ({ testId, isExpanded, type, data, content }) => {
+  const listRef = useRef(null);
+  const [listHeight, setListHeight] = useState();
+
+  const contentWithDefaults = useMemo(() => defaultsDeep({}, content, defaultProps.content), [
+    content,
+  ]);
+
+  const options = getChartOptions(type, data.length > 1, contentWithDefaults);
+
+  useEffect(() => {
+    setListHeight(listRef?.current?.clientHeight);
+  }, [listRef]);
```

Comment:
Are we adding this ref to just get rid of the linter warning? Just want to be sure we know that changes to refs do not cause a component to re-render. If you want to re-render when the height changes we have to put `listRef.current?.clientHeight` in the dependency array. 
```suggestion
  }, [listRef.current?.clientHeight]);
```